custom_components/mikrotik_router/mikrotik_controller.py

```diff
@@ -634,3 +665,229 @@ def get_queue(self):
             self.data["queue"][uid]["upload-burst-time"] = upload_burst_time
             self.data["queue"][uid]["download-burst-time"] = download_burst_time
+
+    # ---------------------------
+    #   get_dhcp
+    # ---------------------------
+    def get_dhcp(self):
+        """Get DHCP data from Mikrotik"""
+
+        self.data["dhcp-server"] = parse_api(
+            data=self.data["dhcp-server"],
+            source=self.api.path("/ip/dhcp-server"),
+            key="name",
+            vals=[
+                {"name": "name"},
+                {"name": "interface", "default": ""},
+            ]
+        )
+
+        # Build list of local DHCP networks
+        dhcp_networks = parse_api(
+            data={},
+            source=self.api.path("/ip/dhcp-server/network"),
+            key="address",
+            vals=[
+                {"name": "address"},
+            ],
+            ensure_vals=[
+                {"name": "address"},
+            ]
+        )
+        self.local_dhcp_networks = [IPv4Network(network) for network in dhcp_networks]
+
+        self.data["dhcp"] = parse_api(
+            data=self.data["dhcp"],
+            source=self.api.path("/ip/dhcp-server/lease"),
+            key="mac-address",
+            vals=[
+                {"name": "mac-address"},
+                {"name": "address", "default": "unknown"},
+                {"name": "host-name", "default": "unknown"},
+                {"name": "status", "default": "unknown"},
+                {"name": "last-seen", "default": "unknown"},
+                {"name": "server", "default": "unknown"},
+                {"name": "comment"},
+            ],
+            ensure_vals=[
+                {"name": "interface"},
+                {"name": "available", "type": "bool", "default": False},
+            ]
+        )
+
+        for uid in self.data["dhcp"]:
+            self.data["dhcp"][uid]['interface'] = \
+                self.data["dhcp-server"][self.data["dhcp"][uid]['server']]["interface"]
+
+            self.data["dhcp"][uid]['available'] = \
+                self.api.arp_ping(self.data["dhcp"][uid]['address'], self.data["dhcp"][uid]['interface'])
+
+    def _address_part_of_local_network(self, address):
+        address = ip_address(address)
+        for network in self.local_dhcp_networks:
+            if address in network:
+                return True
+        return False
+
+    def _get_accounting_mac_by_ip(self, requested_ip):
+        for mac, vals in self.data['accounting'].items():
+            if vals.get('address') is requested_ip:
+                return mac
+        return None
+
+    def get_accounting(self):
+        """Get Accounting data from Mikrotik"""
+        # Check if accounting and account-local-traffic is enabled
+        accounting_enabled, local_traffic_enabled = self.api.is_accounting_and_local_traffic_enabled()
+        traffic_type, traffic_div = self._get_traffic_type_and_div()
+
+        if not accounting_enabled:
```

Comment:
we could use entity availability here instead using sensor class property.
0 is a value, but if we set entity as unavailable, it will change to gray and wont submit value data to database or automation.
```
    @property
    def available(self) -> bool:
        """Return True if entity is available."""
        return True
```

Reply:
most simple would be probably to store sensor availability in data["accounting"]